Sources/SwiftGraphQLUtils/Extensions/String+Case.swift

```diff
@@ -73,9 +73,21 @@ extension String {
         return result
     }
 
-    /// Returns the string camelCased.
-    public var camelCase: String {
+    /// Returns the string – camelCased – while retaining all leading and trailing underscores
+    ///
+    ///     _foo_             // returns _foo_
+    ///     ___foo_bar___     // returns ___fooBar___
+    ///     
+    public var camelCasePreservingSurroundingUnderscores: String {
+        let leading = prefix { $0 == "_" }
+        let remainder = dropFirst(leading.count)
+        let trimmed = remainder.trimmingCharacters(in: CharacterSet(["_"]))
+        let trailing = String.SubSequence(repeating: "_", count: remainder.count - trimmed.count)
```

Comment:
There could be a better way to do this that results in less copies, happy for input here. I just put something together to prove/disprove the issue I was seeing.

Essentially I'm trimming the underscores, while keeping track of the count difference before and after trimming the suffix, so I know how many to put back. The prefix is simpler using the `prefix(while:)` API